rethinkdb/datadog_checks/rethinkdb/operations.py

```diff
@@ -0,0 +1,293 @@
+        .merge({'server': server_config.get(r.row['replica']['server'])})
+        .filter(r.row['server'])  # Skip replicas stored on disconnected servers.
+        .merge({'server': r.row['server'].pluck('id', 'name', 'tags')})
+        # Grab statistics for each replica.
+        # See: https://rethinkdb.com/docs/system-stats/#replica-tableserver-pair
+        .merge(
+            {
+                'stats': stats.get(['table_server', r.row['table']['id'], r.row['server']['id']]).pluck(
+                    'query_engine', 'storage_engine'
+                ),
+            }
+        )
+    )
+
+    rows = query.run(conn)  # type: Iterator[Mapping[str, Any]]
+
+    for row in rows:
+        table = row['table']  # type: Table
+        server = row['server']  # type: Server
+        replica = row['replica']  # type: ShardReplica
+        replica_stats = row['stats']  # type: ReplicaStats
+
+        tags = [
+            'table:{}'.format(table['name']),
+            'database:{}'.format(table['db']),
+            'server:{}'.format(server['name']),
+            'state:{}'.format(replica['state']),
+        ]
+        tags.extend(server['tags'])
+
+        yield replica_stats, tags
+
+
+def get_table_statuses(conn, **kwargs):
+    # type: (rethinkdb.net.Connection, **Any) -> Iterator[Tuple[TableStatus, List[str]]]
+    """
+    Retrieve the status of each table in the cluster.
+    """
+    for table_status in system.table('table_status').run(conn):  # type: TableStatus
+        tags = ['table:{}'.format(table_status['name']), 'database:{}'.format(table_status['db'])]
+        yield table_status, tags
+
+
+def get_server_statuses(conn, **kwargs):
+    # type: (rethinkdb.net.Connection, **Any) -> Iterator[Tuple[ServerStatus, List[str]]]
+    """
+    Retrieve the status of each server in the cluster.
+    """
+    for server_status in system.table('server_status').run(conn):  # type: ServerStatus
+        tags = ['server:{}'.format(server_status['name'])]
+        yield server_status, tags
+
+
+def get_system_jobs(conn, config, **kwargs):
+    # type: (rethinkdb.net.Connection, Config, **Any) -> Iterator[Tuple[Job, List[str]]]
+    """
+    Retrieve all the currently running system jobs.
+    """
+    for job in system.table('jobs').run(conn):  # type: Job
+        tags = ['job_type:{}'.format(job['type'])]
+        tags.extend('server:{}'.format(server) for server in job['servers'])
+
+        # Follow job types listed on: https://rethinkdb.com/docs/system-jobs/#document-schema
+
+        if job['type'] == 'query':
+            # NOTE: we can only consistently collect metrics about queries that span more than an Agent collection
+            # interval. (There will be many short-lived queries within two checks that we can't capture.)
+            # Here, this means only changefeed queries and abnormally long request-response queries will pass through.
+            if job['duration_sec'] < config.min_collection_interval:
```

Comment:
> we can only consistently collect metrics about queries that span more than an Agent collection interval.

1. I'm not sure to understand why ☝️  and how `duration_sec ` relates to `min_collection_interval`.

2. Are the jobs retrieved ephemeral/sparse ? If so, I'm not sure how this condition will `if job['duration_sec'] < config.min_collection_interval` solve the issue.

Reply:
> Are the jobs retrieved ephemeral/sparse

Yes, the `jobs` system table only contains _actively running_ jobs.

For example, assuming the `min_collection_interval` is D seconds, and a first check occurs at time T, then if a query starts a T + 1 and finishes at T + (D - 1), and a second check runs at T + D, then that query won't be captured by the Agent (correct me if I'm missing something). The goal is to not report any such queries because it would be noisy/inconsistent (we may get some queries but not others).

Do you have wording suggestions for the comment that would sound clearer to you.

Reply:
> Yes, the jobs system table only contains actively running jobs.

This will make this metric quite unstable even with `if job['duration_sec'] < config.min_collection_interval`, since job duration vary each time the job runs.

I think this is better addressed by logs / slow logs. If the logs for job duration is not available, maybe we can submit a PR to log it.

Reply:
Indeed. Slow logs aren't supported yet but there's a long-open issue upstream - https://github.com/rethinkdb/rethinkdb/issues/2238

Okay, so let's just drop query jobs for now?

Reply:
Dropped in f03beb071

Reply:
Yes. I think it's better to not support it for now than having something that might be not reliable.

If user request it, we can put effort to help slow logs being implemented upstream.

Reply:
Dropped these duration metrics in favor of a jobs count summary - 7d962c772464bc9249e34debe09efc98ecae99ed